### PR TITLE
Prepare atomicfu build scripts for inclusion into the main aggregate as a user project

### PR DESCRIPTION
**Pull request by anastasiiaSpaseeva**

### CODE REVIEW

File: atomicfu-gradle-plugin/build.gradle [hunk 1]
```diff
@@ -14,8 +14,8 @@ if (rootProject.ext.jvm_ir_enabled) {
 // Gradle plugin must be compiled targeting the same Kotlin version as used by Gradle
 kotlin.sourceSets.all {
     languageSettings {
-        apiVersion = "1.4"
-        languageVersion = "1.4"
+        apiVersion = rootProject.ext.kotlin_api_version
```
Comment (qwwdfsad) [inline atomicfu-gradle-plugin/build.gradle:17]: This is a bit too unmaintainable for us -- both in terms of readability (i.e. _why_ the code is written this way) and writability (i.e. where to update the language version).

I suggest making rr build explicit:
```
languageSettings {
    apiVersion = getApiVersionForOverriddenKotlin() ?: "1.4"
    ... same pattern everywhere else...
}
```

This way we know **why** the version can be overridden, while still having the actual default where it is supposed to be

File: atomicfu-maven-plugin/build.gradle [hunk 2]
```diff
@@ -23,6 +23,10 @@ def buildSnapshots = rootProject.properties['build_snapshot_train'] != null
 
 evaluationDependsOn(':atomicfu-transformer')
 
+def kotlinDevRepoUrl = (rootProject.properties['kotlin_repo_url'] ?: "" ) as String
+println("Configured external repository url: '$kotlinDevRepoUrl'")
```
Comment (qwwdfsad) [inline atomicfu-maven-plugin/build.gradle:27]: Please avoid `println` calls enabled by default

File: atomicfu-maven-plugin/build.gradle [hunk 3]
```diff
@@ -23,6 +23,10 @@ def buildSnapshots = rootProject.properties['build_snapshot_train'] != null
 
 evaluationDependsOn(':atomicfu-transformer')
 
+def kotlinDevRepoUrl = (rootProject.properties['kotlin_repo_url'] ?: "" ) as String
```
Comment (qwwdfsad) [inline atomicfu-maven-plugin/build.gradle:26]: We should have a comment what this is for

File: build.gradle [hunk 4]
```diff
@@ -28,13 +28,15 @@ buildscript {
     // These two flags are enabled in train builds for JVM IR compiler testing
     ext.jvm_ir_enabled = rootProject.properties['enable_jvm_ir'] != null
     ext.native_targets_enabled = rootProject.properties['disable_native_targets'] == null
+    def kotlinDevRepoUrl = (rootProject.properties['kotlin_repo_url'] ?: "" ) as String
```
Comment (qwwdfsad) [inline build.gradle:31]: Same here

**Review by qwwdfsad — CHANGES_REQUESTED** [review]
* Please change the target branch to `develop`
* Currently, all RR/train support is stretched through all Gradle files. It's both hard to maintain and understand. Please create a separate `.kts` file and declare all RR-specific variables and functions there. Then on call-sites we will have clear entry-point:

```
// KotlinDevSupport.kts
// Here will be a huge comment explaining why this is needed and how it is used

val kotlinDevRepo = ...
fun getApiVersionForOverriddenKotlin() = ...
```

Pros: everything is documented in a single place, easy to maintain, hard to miss, no questions on individual call sites

File: atomicfu-maven-plugin/build.gradle (hunk 3 above)
Comment (anastasiiaSpaseeva) [reply atomicfu-maven-plugin/build.gradle]: I've added a kdoc in KotlinAggregateBuild.kt

**Review by qwwdfsad — APPROVED** [review]

**Pull request merged**